Tasks/DeployAzureResourceGroup/AzureResourceManagerHelper.ps1

```diff
@@ -674,3 +671,73 @@ function Print-OperationLog
         }
     }
 }
+
+function Delete-MachineGroupFromProvider
+{
+    param([string]$machineGroupName)
+
+    Write-Host (Get-LocalizedString -Key "[Azure Resource Manager]Deleting resource group {0}" -ArgumentList $machineGroupName)
+
+    Remove-AzureResourceGroup -ResourceGroupName $machineGroupName -Force -ErrorAction Stop -Verbose
+    Write-Host (Get-LocalizedString -Key "[Azure Resource Manager]Deleted resource group '{0}' from Azure provider" -ArgumentList $machineGroupName)
+}
+
+function Delete-MachineFromProvider
```

Comment:
Do you want to use 'Azure' explicitly instead of 'Provider' in the name?
Because the method is not actually generic enough. 
Further, the logs and warning messages are calling out "Azure"